src/components/SideNavigation/SideNavigation.tsx

```diff
@@ -0,0 +1,140 @@
+
+import * as React from 'react';
+import { themr, ThemedComponentClass } from 'react-css-themr';
+import Icon from '../Icon';
+import { SIDENAVIGATION } from '../ThemeIdentifiers';
+import { Drawer, DrawerContent } from '../Drawer';
+import Button from '../Button';
+import Tooltip from '../Tooltip';
+import Accordion from '../Accordion';
+import {AccordionItemProps} from '../Accordion';
+import * as baseTheme from './SideNavigation.scss';
+
+export interface INavigationData {
+    id?:number;
+    label?: React.ReactNode;
+    icon?: React.ReactNode;
+    children?: React.ReactNode;
+    action?(): void;
```

Comment:
@ishani-m Action can also accept any argument & then send it back in callback function so please add that & make is optional its not necessary that user will want that everytime